src/main/java/com/sarahhand/fractals/view/FractalsUI.java

```diff
@@ -261,60 +333,141 @@ public void actionPerformed(ActionEvent ae) {
+			List<ColorScheme> supportedColorSchemes = currentEvents.getFractalViewer().getSupportedColorSchemes();
 			for(int count = 0; count < supportedColorSchemes.size(); count++) {
 				if(supportedColorSchemes.get(count).getName().equals(colorSchemeName)) {
-					colorScheme = events.getFractalViewer().getSupportedColorSchemes().get(count);
+					colorScheme = currentEvents.getFractalViewer().getSupportedColorSchemes().get(count);
 				}
 			}
 		}
 
 		public void actionPerformed(ActionEvent ae) {
-			events.setColorScheme(colorScheme);
-			image.setImage(events.getFractalViewer().getView(frameDimension));
+			currentEvents.setColorScheme(colorScheme);
+			image.setImage(currentEvents.getFractalViewer().getView(frameDimension));
 			frame.repaint();
 		}
 	}
 
+	/** This serves as the ActionListener for the setJuliaConstantsMenuItem JMenuItem. It brings up a window that
+	 * let's the user change the Julia Set constants. 
+	 * 
+	 * @author M00031
+	 *
+	 */
+	private class SetJuliaConstantsMenuItemActionListener implements ActionListener {
+
+		public void actionPerformed(ActionEvent ae) {
+			ChangeJuliaConstants changeConstants = new ChangeJuliaConstants(getThis());
+			Thread changeConstantsThread = new Thread(new ChangeConstantsRunnable(changeConstants));
+			changeConstantsThread.start();
+		}
+	}
+
 	/** This lets the ChangeColorPalette frame run on a different Thread.
 	 * 
 	 * @author M00031
 	 *
 	 */
-	private class ChangePaletteRunnable implements Runnable {
+	private class ChangeConstantsRunnable implements Runnable {
 
-		private ChangeColorPalette changeCP;
+		private ChangeJuliaConstants changeConstants;
 
-		public ChangePaletteRunnable(ChangeColorPalette changeCP) {
-			this.changeCP = changeCP;
+		public ChangeConstantsRunnable(ChangeJuliaConstants changeConstants) {
+			this.changeConstants = changeConstants;
 		}
 
 		public void run() {
-			while (changeCP.getCreatedColorPalette() == null) {}
-			events.setColorPalette(changeCP.getCreatedColorPalette());
-			image.setImage(events.getFractalViewer().getView(frameDimension));
+			while(changeConstants.isFinished() == false) {}
+			juliaEvents.getFractalViewer().getConfig().setConstants(changeConstants.getJuliaConstants());
+			currentEvents = juliaEvents;
+			image.setImage(currentEvents.getFractalViewer().getView(frameDimension));
 			frame.repaint();
 		}
 	}
-	
+
+	/** This class serves as the ActionListener for the mandelbrotSetMenuItem JMenuItem. It switches to
+	 * viewing the Mandelbrot Set.
+	 * 
+	 * @author M00031
+	 *
+	 */
+	private class MandelbrotSetMenuItemActionListener implements ActionListener {
+
+		public void actionPerformed(ActionEvent ae) {
+			imageLabel.removeMouseListener(currentEvents);
+			imageLabel.removeMouseMotionListener(currentEvents);
+			currentEvents = mandelbrotEvents;
+			setSelectedColorScheme();
+			createColorPaletteMenuItem.setEnabled(false);
```

Comment:
Actually, I got rid of this in my local code. The Julia Set also has ColorPalettes, so it is not needed.